### DISCUSSION TITLE
Codec Adapter Discussion Topics

### DISCUSSION THREAD

**Discussion by cujomalainey** [thesofproject/sof#7389] [Team Posts]
Please feel free to edit and add items. Meeting set for ~2/25, check marks depict resolved topics

- [ ] Double extra state machine, still unclear on why this is needed
- [x] Config calls not always going through
  - Discussed and decided to allow INIT state calls to go through, rather than just prepare or higher.
- [x]  Why are we managing memory on top of memory
  - Discussed and will attempt to migrate into component layer
- [ ] Limited types of configs to be passed
- [ ] Why not call apply config for all types of config?
- [ ] Global blob size restriction vs defined by codec?
- [ ] Registration system
  - Move to either Linker based like component or function register
- [ ] Cadence prepare flow. Some algorithms (e.g mp3) need to parse the header in order to find out the config parameters (e.g sample rate, bitrate, etc).

**Comment by cujomalainey** [comment]
@plbossart did you want to add something about the ID system?

**Comment by lgirdwood** [comment]
All, please select your availability
https://doodle.com/poll/cfkefycsvsqkh6wx?utm_source=poll&utm_medium=link

@stolx @worldgeek @ryans-lee please consider attending if you can. We are discussing codec adapter improvements and it would be good to hear ideas from your experiences and perspective.

**Comment by lgirdwood** [comment]
and @mrajwa too, please select your availability.

**Comment by lgirdwood** [comment]
@plbossart @harshapriya-n @beatabaranowska @mwasko @cujomalainey @dgreid @bzhg @dbaluta fyi

**Comment by cujomalainey** [comment]
might need to send to @mrajwa directly as he is not in this group

**Comment by cujomalainey** [comment]
@lgirdwood do you still want to invite @stolx ?

**Comment by lgirdwood** [comment]
@cujomalainey yes, if it's OK with stolx and everyone.

**Comment by lgirdwood** [comment]
Looks like Thursday 25th 5 - 6pm GMT will send out invite. - 7 votes.

**Comment by dbaluta** [comment]
@lgirdwood yes, this looks to be the best fit. I wonder if @mrajwa can somehow join. He is the only one with cannot attend option.

![image](https://user-images.githubusercontent.com/62316/108756469-352eea80-7551-11eb-8c73-208f0db6ff35.png)

**Comment by mrajwa** [comment]
@dbaluta I will adjust my calendar to join this meeting. 
In the meantime, what does " Limited types of configs to be passed" mean? What types of config are not supported as of today?

**Comment by dbaluta** [comment]
@dbaluta I'm not sure I understand what that does mean. It was not added by me. Maybe @cujomalainey knows. 
Altough, I do suspect that there might be specific parameters for specific codecs.

**Comment by lgirdwood** [comment]
@juimonen @mrajwa @dbaluta can we update the current progress here. I think we have some items that are now merged or have PRs pending.
